### PR TITLE
xkb-switch-i3: init at 1.8.1

### PR DESCRIPTION
**Pull request by ewok-old**
###### Motivation for this change
Adding xkb-switch with i3 support

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2863,6 +2863,12 @@
     githubId = 30512529;
     name = "Evils";
   };
+  ewok = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:2866]: Please put this into a separate commit with the message ``maintainers: add ewok``.

File: pkgs/tools/X11/xkb-switch-i3/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+, libX11
+, libxkbfile
+, pkg-config
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xkb-switch-i3";
+  version = "1.8.1";
+
+  src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/tools/X11/xkb-switch-i3/default.nix:16]: Please use fetchFromGitHub.

File: pkgs/tools/X11/xkb-switch-i3/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    url = "https://github.com/Zebradil/xkb-switch-i3";
+    rev = version;
+    sha256 = "15c19hp0n1k3w15qn97j6wp5b8hbk0mq6x3xjfn6dkkjfz1fl6cn";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/tools/X11/xkb-switch-i3/default.nix:20]: I would prefer if we add a package for https://github.com/drmgc/i3ipcpp instead of using the vendored version to more easily patch it.

File: pkgs/tools/X11/xkb-switch-i3/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ i3 jsoncpp libsigcxx libX11 libxkbfile pkg-config ];
```
Comment (SuperSandro2000) [inline pkgs/tools/X11/xkb-switch-i3/default.nix:24]:
```suggestion
  nativeBuildInputs = [ cmake pkg-config ];
  buildInputs = [ i3 jsoncpp libsigcxx libX11 libxkbfile ];
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107579` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xkb-switch-i3</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (ewok) [reply maintainers/maintainer-list.nix:2866]: This is done

File: pkgs/tools/X11/xkb-switch-i3/default.nix (hunk 2 above)
Comment (ewok) [reply pkgs/tools/X11/xkb-switch-i3/default.nix]: I've changed that, thx

File: pkgs/tools/X11/xkb-switch-i3/default.nix (hunk 4 above)
Comment (ewok) [reply pkgs/tools/X11/xkb-switch-i3/default.nix]: Changed that too, thank you

File: pkgs/tools/X11/xkb-switch-i3/default.nix (hunk 3 above)
Comment (ewok) [reply pkgs/tools/X11/xkb-switch-i3/default.nix:21]: I am not sure yet how to do that. I am confused by the message from the repo:
> Yet the only way of using is to add this repo as a submodule

Comment (SuperSandro2000) [reply pkgs/tools/X11/xkb-switch-i3/default.nix:21]: Then we just keep it this way. I thought this would be easier.

File: pkgs/tools/X11/xkb-switch-i3/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ cmake pkg-config ];
+  buildInputs = [ i3 jsoncpp libsigcxx libX11 libxkbfile ];
+
+  meta = with stdenv.lib; {
+    description = "Switch your X keyboard layouts from the command line(i3 edition)";
+    homepage = "https://github.com/Zebradil/xkb-switch-i3";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/tools/X11/xkb-switch-i3/default.nix:30]: `gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107579` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xkb-switch-i3</li>
  </ul>
</details>

File: pkgs/tools/X11/xkb-switch-i3/default.nix (hunk 5 above)
Comment (ewok) [reply pkgs/tools/X11/xkb-switch-i3/default.nix]: Updated, thx!

**Review by SuperSandro2000 — APPROVED** [review]
